tensorzero-core/src/variant/dicl.rs

```diff
@@ -45,28 +45,138 @@ use super::{
 /// We need a helper to deserialize the config because it relies on
 /// a path to a file for system instructions and we need to use the
 /// load() step to get the fully qualified path.
-#[derive(Debug, Default, Serialize, Deserialize)]
+#[derive(Debug, Default, Serialize)]
 #[cfg_attr(test, derive(ts_rs::TS))]
 #[cfg_attr(test, ts(export))]
 pub struct DiclConfig {
-    pub weight: Option<f64>,
-    pub embedding_model: Arc<str>,
-    pub k: u32, // k as in k-nearest neighbors
-    pub model: Arc<str>,
-    pub system_instructions: String,
-    pub temperature: Option<f32>,
-    pub top_p: Option<f32>,
-    pub stop_sequences: Option<Vec<String>>,
-    pub presence_penalty: Option<f32>,
-    pub frequency_penalty: Option<f32>,
-    pub max_tokens: Option<u32>,
-    pub seed: Option<u32>,
-    pub json_mode: Option<JsonMode>,
+    weight: Option<f64>,
+    embedding_model: Arc<str>,
+    k: u32, // k as in k-nearest neighbors
+    model: Arc<str>,
+    system_instructions: String,
+    temperature: Option<f32>,
+    top_p: Option<f32>,
+    stop_sequences: Option<Vec<String>>,
+    presence_penalty: Option<f32>,
+    frequency_penalty: Option<f32>,
+    max_tokens: Option<u32>,
+    seed: Option<u32>,
+    json_mode: Option<JsonMode>,
     #[cfg_attr(test, ts(skip))]
-    pub extra_body: Option<ExtraBodyConfig>,
+    extra_body: Option<ExtraBodyConfig>,
     #[cfg_attr(test, ts(skip))]
-    pub extra_headers: Option<ExtraHeadersConfig>,
-    pub retries: RetryConfig,
+    extra_headers: Option<ExtraHeadersConfig>,
+    retries: RetryConfig,
+}
+
+impl DiclConfig {
+    /// Create a new DiclConfig - this is the only way to construct one from outside the module
+    #[expect(clippy::too_many_arguments)]
```

Comment:
can we make this `#[cfg(test)]` or remove it if unused?
that would require going through `UninitializedDiclConfig::load()` in the evaluator but I think that's actually good practice and the primary purpose of this refactor (always using load()).